ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -658,6 +660,30 @@ class ClientQueryViewController: UITableViewController, Themeable {
 		}
 
 	}
+
+	func openInRow(_ item: OCItem, viewDidAppearHandler: ClientActionVieDidAppearHandler? = nil, completionHandler: ClientActionCompletionHandler? = nil) {
+
+		if let progress = self.core?.downloadItem(item, options: nil, resultHandler: { (error, _, _, file) in
+			if error != nil {
```

Comment:
@pablocarmu If I remember correctly completion handler of` downloadItem()` is not guaranteed to be called on the main thread, so are we missing dispatch on main thingy here?

Reply:
@pablocarmu I guess I had to wait until you remove [WIP] to start reviewing..